README.md

```diff
@@ -1,94 +1,113 @@
+### Install from GitHub
 
-# Minimum install (no optional dependencies, good for Raspberry PI builds).
-python -m pip install --force-reinstall git+https://github.com/braingeneers/braingeneerspy.git
+```bash
+pip install --force-reinstall git+https://github.com/braingeneers/braingeneerspy.git
 ```
 
-### macOS installation note:
-if install fails with ```no matches found: git+https://github.com/braingeneers/braingeneerspy.git#egg=braingeneerspy[all]```
-wrap quotes around the github address like so 
+### Install from a Wheel (PyPI)
 
+If you prefer to install a pre-built wheel due to resource limitations, you can find the latest release on [PyPI](https://pypi.org/project/braingeneers/).
+
+```bash
+pip install braingeneerspy.whl
 ```
-# Typical install (includes `iot`, `analysis`, and `data` access functions, skips `ml`, and lab-specific dependencies): 
-python -m pip install --force-reinstall 'git+https://github.com/braingeneers/braingeneerspy.git#egg=braingeneerspy[iot,analysis]'
 
-# Full install (all optional dependencies included).
-python -m pip install --force-reinstall 'git+https://github.com/braingeneers/braingeneerspy.git#egg=braingeneerspy[all]'
+### Install with Optional Dependencies
+
+You can install `braingeneerspy` with specific optional dependencies based on your needs. Use the following command examples:
+
+- Install with IoT, analysis, and data access functions (skips machine learning and lab-specific dependencies):
+
+```bash
+pip install --force-reinstall git+https://github.com/braingeneers/braingeneerspy.git#egg=braingeneerspy[iot,analysis,data]
 ```
 
-### Optional dependency organization
+- Install with all optional dependencies:
+
+```bash
+pip install --force-reinstall git+https://github.com/braingeneers/braingeneerspy.git#egg=braingeneerspy[all]
+```
+
+Please note that macOS users may need to wrap the GitHub URL in quotes if they encounter issues during installation, as shown in the examples above.
+
+## Optional Dependency Groups
+
+Dependencies in `braingeneerspy` are organized into optional groups of requirements. You can install all dependencies with `all`, or you can install a specific set of dependencies. Here are the optional dependency groups:
+
+- *Unspecified*: Minimal packages for data access will be installed.
+- `all`: All optional dependencies will be included.
+- `iot`: IoT dependencies such as AWS and Redis packages will be installed.
+- `analysis`: Dependencies for data analysis routines, plotting tools, math libraries, etc.
+- `ml`: Machine learning dependencies such as `torch` will be installed.
+- `hengenlab`: Hengenlab data loader-specific packages such as `neuraltoolkit` will be installed.
+
+## Committing Changes to the Repo
+
+If you plan to make changes to the `braingeneerspy` package and publish them on GitHub, please follow these steps:
+
+1. Update the `version` variable in `setup.py`.
```

Comment:
This is old instruction text. Not needed anymore.

Reply:
Updated to CONTRIBUTING.md